### PR TITLE
Feature/user dto

### PR DESCRIPTION
**Pull request by mazanec202**

### CODE REVIEW

File: api/src/main/java/tennisclub/dto/UserAuthDTO.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package tennisclub.dto;
+
+public class UserAuthDTO {
+
+    private  String username;
```
Comment (Fuyunori) [inline api/src/main/java/tennisclub/dto/UserAuthDTO.java:5]: I think that when the user is authenticating, he/she should always provide a username and password - they shouldn't be blank. Therefore I would personally add a `@NotBlank` annotation 😄

**Review by Fuyunori — COMMENTED** [review]
Shouldn't there also be a DTO when a user is registering? 🤔 

Anyways, great job and thank you😄

File: api/src/main/java/tennisclub/dto/UserAuthDTO.java (hunk 1 above)
Comment (mazanec202) [reply api/src/main/java/tennisclub/dto/UserAuthDTO.java:5]: Oh, completely forgot the validation part, thanks!

**Review by Fuyunori — APPROVED** [review]

**Comment by Fuyunori** [comment]
I am only missing UserRegisterDTO (or something of that sort), otherwise it should be okay 😄

File: api/src/main/java/tennisclub/dto/UserAuthDTO.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    public String getPasswordHash() {
+        return password;
+    }
+
+    public void setPasswordHash(String password) {
+        this.password = password;
+    }
+}
```
Comment (Fuyunori) [inline api/src/main/java/tennisclub/dto/UserAuthDTO.java:30]: I overlooked this. It is missing `equals/hashCode` methods 👀

**Comment by mazanec202** [comment]
> I am only missing UserRegisterDTO (or something of that sort) due to having a register method on the service layer, otherwise it should be okay 😄

My thoughts were that the registration would be a two-step process (using some sort of stepper UI component): 
1) register with username & password (check that username is not taken, the password is valid (which reminds me to add at least some length verification) etc) ASAP
2) fill in the optional details

In my opinion, this would make the registration a bit more clear, however, I'm open for discussion :)

**Comment by Fuyunori** [comment]
@mazanec202 I see - so when creating a new User, you're going to use the UserAuthDTO? 🤔

**Comment by mazanec202** [comment]
> @mazanec202 I see - so when creating a new User, you're going to use the UserAuthDTO? 🤔

Precisely, that was my intention.

**Comment by Fuyunori** [comment]
@mazanec202 Thank you for the clarification! 😄

**Pull request merged**